hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMDirectoriesPurgeRequestWithFSO.java

```diff
@@ -75,12 +75,13 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
                 volumeName, bucketName);
             lockSet.add(volBucketPair);
           }
-          OmBucketInfo omBucketInfo = getBucketInfo(omMetadataManager,
-              volumeName, bucketName);
+          updateBucketInfo(volBucketInfoMap, omMetadataManager, path,
+              volumeName, bucketName, volBucketPair);
+          OmBucketInfo omBucketInfo = volBucketInfoMap.get(volBucketPair);
```

Comment:
Thanks @sumitagrawl for the steps to reproduce.  With the patch, what happens if the `(vol,bucket)` pair are in the `volBucketInfoMap` and _during_ the recursive delete the bucket is re-created and deleted multiple times before the first delete has completed?

Reply:
1. This Request is handled in Ratis Transaction which is synchronized flow, other write operation like create/delete bucket needs wait till this is completed.. So while this map is populated, parallel create or delete is not allowed in this flow. This is important for write operation to have consistent data.
2. If bucket or volume is removed before this operation start, then bucketInfo will be null in this case and namespace/size quota will not be updated
3. If other bucket is created, then here as fix, we are not updating to new bucketInfo.